mlx/array.cpp

```diff
@@ -47,6 +47,17 @@ array::array(
           std::move(primitive),
           inputs)) {}
 
+array::array(
+    std::vector<int> shape,
+    Dtype dtype,
+    std::shared_ptr<Primitive> primitive,
+    std::vector<array>&& inputs)
+    : array_desc_(std::make_shared<ArrayDesc>(
+          std::move(shape),
+          dtype,
+          std::move(primitive),
+          std::move(inputs))) {}
```

Comment:
@angeloskath @jagrit06  this is unrelated to compile, but I added a move constructor for the array. It's pretty minor but we should start moving to it in general. There are a lot of cases where we make deep copies of the inputs vector  and shape vector which could be trivially avoided (it actually can make a difference for lot s of small ops on CPU):

Note the interface you can optionally move the shape into it but it only takes an r-value for the inputs (otherwise it's ambiguous with the `const&` constructor.

But we have a lot of cases where you want to deep copy the shape and move the inputs, like this:
```
array(shape, type, primitive, std::move(inptus));
```

Or just move everything if you don't need the shape anymore:

```
array(std::move(shape), type, primitive, std::move(inptus));
```

I couldn't find cases where you want to copy the inputs and move the shape.. so I didn't add a constructor for that.